src/api/ucc.h

```diff
@@ -0,0 +1,709 @@
+
+
+/**
+ *  @ingroup UCC_TEAM
+ *
+ *  @brief @ref ucc_team_create_test routines tests the status of team handle.
+ *  If required it can progress the communication but cannot block on the
+ *  communications.
+ *
+ *  @param  [in] ucc_team  Team handle to test
+ *
+ * @return Error code as defined by @ref ucc_status_t
+ */
+
+ucc_status_t ucc_team_create_test();
+
+
+/**
+ * @ingroup UCC_TEAM
+ *
+ * @brief ucc_team_destroy is a blocking collective operation to release all
+ * resources associated with the team handle, and destroy the team handle. It is
+ * invalid to post a collective operation after the ucc_team_destroy operation.
+ *
+ * @param  [in] team  Destroy previously created team and release all resources
+ *                     associated with it.
+ *
+ * @return Error code as defined by @ref ucc_status_t
+ *
+ */
+
+ucc_status_t ucc_team_destroy(ucc_team_h team);
+
+/**
+ * @ingroup UCC_TEAM
+ * @brief @ref ucc_team_get_attribs routine queries the team handle
+ * attributes. The attributes of the team handle are described by the team
+ * attributes @ref ucc_team_attrib_t
+ *
+ *  @param [in]   ucc_team       Team handle 
+ *  @param [out]  team_attribs   Attributes of the team
+ *
+ *  @return Error code as defined by @ref ucc_status_t
+ */
+
+ucc_status_t ucc_team_get_attribs(ucc_team_h ucc_team, ucc_team_attribs_t
+                                  *team_atribs);
+
+/**
+ * @ingroup UCC_TEAM
+ *
+ * @brief ucc_team_create_from_parent is a nonblocking collective operation,
+ * which creates a new team from the parent team. If a participant intends to
+ * participate in the new team, it passes a TRUE value for the “included”
+ * parameter. Otherwise, it passes FALSE. The routine returns immediately after
+ * the post-operation. To learn the completion of the team create operation, the
+ * ucc_team_create_test operation is used. 
+
+ * @param [in]     my_ep         Endpoint of the process/thread calling the split operation
+ * @param [in]    parent_team    Parent team handle from which a new team handle is created
+ * @param [in]    included       Boolean variable indicating whether the
+ *                               process/thread participates in the newly created team
+ * @param [out]    new_ucc_team   Pointer to the new team handle
+ *
+ * @return Error code as defined by @ref ucc_status_t
+ */
+
+ucc_status_t ucc_team_create_from_parent(uint64_t my_ep, bool included,
+                                             ucc_team_h parent_team, 
+                                             ucc_team_h *new_ucc_team);
+
+/**
+ * @ingroup UCC_TEAM
+ *
+ * @ref ucc_team_get_size routine queries the size of the team. It reflects the
+ * number of unique endpoints in the team.
+ *
+ * @param [in]   ucc_team  Team handle
+ * @param [out]  size      The size of team as number of endpoints
+ *
+ * @return Error code as defined by @ref ucc_status_t
+ */
+
+ucc_status_t ucc_team_get_size(ucc_team_h ucc_team, uint32_t *size);
```

Comment:
Should we make team size part of team attributes?

Reply:
Same here. It is more of a convenient interface for frequently used functionality.